punx/h5tree.py

```diff
@@ -320,10 +319,14 @@ def __render(obj, rank, key, indents):
             elif rank < 4:
                 # this replaces a lot of code: if rank == ...
                 indices = ", ".join([str(key)] + (":" * (rank - 1)).split())
-                part = eval("obj[%s]" % indices)
+                try:
+                    part = eval(f"obj[{indices}]")
+                except OSError as exc:
```

Comment:
Curious: what causes an OSError when trying to index this object?

Reply:
An OSError is raised when the HDF5 library cannot find a compression codec it expects.

Reply:
This error is avoidable with user action (installing the codec package), so it should be accompanied by a UserWarning from the warnings module.

Reply:
Something like `UserWarning("An error occurred when reading a dataset; you probably don't have the proper HDF5 codec installed for this file.")`